src/hotspot/share/gc/z/zPhysicalMemory.cpp

```diff
@@ -46,20 +46,19 @@ ZPhysicalMemory::ZPhysicalMemory()
 
 ZPhysicalMemory::ZPhysicalMemory(const ZPhysicalMemorySegment& segment)
   : _segments() {
-  add_segment(segment);
+  _segments.append(segment);
 }
 
 ZPhysicalMemory::ZPhysicalMemory(const ZPhysicalMemory& pmem)
-  : _segments() {
-  add_segments(pmem);
+  : _segments(pmem.nsegments()) {
+  _segments.appendAll(&pmem._segments);
 }
 
 const ZPhysicalMemory& ZPhysicalMemory::operator=(const ZPhysicalMemory& pmem) {
-  // Free segments
+  // Free and copy segments
```

Comment:
pre-existing: Seems like this ought to be checking for self-assign.  I'd be okay with asserting against, even
though that's often frowned on in various writings on the subject.

Reply:
Checking and returning `*this` seems fine to me. Seems like the least surprising solution to me. 

The copy assignment operator is currently only ever used in a gtest.

Reply:
Agreed that checking, and returning `*this` is the preferable way to handle it.  I offered the assert option in
case there was some concern about doing that.

Reply:
I've added a self-assignment check in a new commit.